beanie/odm/settings/collection.py

```diff
@@ -95,7 +95,8 @@ async def init(
             new_indexes += await collection.create_indexes(found_indexes)
 
         # delete indexes
-        if allow_index_dropping:
+        # Only drop indexes if the user specifically allows for it and Beanie has index fields
+        if allow_index_dropping and found_indexes:
```

Comment:
It doesn't allow full index dropping for people, who manage MongoDB with Beanie only. I mean the case when somebody decides to not use any indexes for the collection more.

Reply:
So my intention was to make this an opt-in process instead of an opt-out process.

There are three cases that I can see:

1. User maintains indexes outside of beanie, no action required and current indexes are maintained
2. User maintains indexes outside of beanie and want to "kick the tires" on beanie creating a new index, the user would add the `index` param to the `Collection` class, no other action required and current indexes are maintained
3. User wants beanie to manage everything and wants beanie to cleanup additional indexes, add `index` param and set `allow_index_dropping` 

In the grand scheme of things having more indexes than you need is not a problem and can easily be taken care of through any of Mongo's tools (compass, atlas, cli, etc), so that is why I wanted to move this to an opt-in on delete.  

Let me know your thoughts on this approach I am happy to change the approach if necessary but my feeling is that deleting indexes should be explicit and opt-in.

Reply:
Sorry for the delay.
Point number 3 doesn't work for the case when there were some indexes in the model before but there are no more. `found_indexes` will be False and indexes will not be deleted even if the user allows it. 

I think the `found_indexes` check is not needed, as the user already allows index dropping explicitly.